packages/client/src/no-op-provider.ts

```diff
@@ -11,6 +11,10 @@ class NoopFeatureProvider implements Provider {
     name: 'No-op Provider',
   } as const;
 
+  get status(): ProviderStatus {
+    return ProviderStatus.NOT_READY;
```

Comment:
Shouldn't the no op provider always be ready?

Reply:
Formerly, the default no-op provider never got ready, because only after calling setProvider the providerReady flag was set to true.
All the tests for the events depend on the ready flag only being called once.
But because `Requirement 2.4.2: Providers without this field can be assumed to be ready immediately ` also for the no-op provider we would have a provider-ready flag which means in the tests, the provider-ready handler is at least called twice.
I thought as the no-op is the "fallback" anyways I would just set it to NOT_READY but well I also see that this may be wrong.
Not sure what to do about this.

Reply:
Would the no op provider ever be set to `READY`? @toddbaert any thoughts on this?

Reply:
Yep, makes sense. A comment would clear up my concerns. Thanks for the update.

Reply:
Added it here @beeme1mr https://github.com/open-feature/js-sdk/pull/429/commits/59621b664d4edbc6a7692070799d9c82c0647fa6# 🙂